src/sudoku.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2023-2024 Vulcalien
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# returns a 'size x size' matrix filled with 'val'
+def generate_square_matrix(size: int, val: any):
+    matrix = []
+    for i in range(size):
+        matrix.append([])
+        for j in range(size):
+            matrix[i].append(val)
+    return matrix
+
+class Sudoku:
+
+    def __init__(self, block_size: int):
+        self.block_size = block_size
+        self.size = pow(block_size, 2)
+
+        self.cells = generate_square_matrix(self.size, None)
+        self.fixed = generate_square_matrix(self.size, False)
+
+    # returns True if 'symbol' can be placed in (x, y), False otherwise
+    def can_set(self, x: int, y: int, symbol: int):
```

Comment:
```suggestion
    def can_set(self, x: int, y: int, symbol: int):
        """returns True if 'symbol' can be placed in (x, y), False otherwise"""
```